mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -952,13 +963,160 @@ private class LogisticAggregator(
     val bcFeaturesStd: Broadcast[Array[Double]],
     private val numFeatures: Int,
     numClasses: Int,
-    fitIntercept: Boolean) extends Serializable {
+    fitIntercept: Boolean,
+    multinomial: Boolean) extends Serializable with Logging {
+
+  private val numFeaturesPlusIntercept = if (fitIntercept) numFeatures + 1 else numFeatures
+  private val coefficientSize = bcCoefficients.value.size
+  if (multinomial) {
+    require(numClasses ==  coefficientSize / numFeaturesPlusIntercept, s"The number of " +
+      s"coefficients should be ${numClasses * numFeaturesPlusIntercept} but was $coefficientSize")
+  } else {
+    require(coefficientSize == numFeaturesPlusIntercept, s"Expected $numFeaturesPlusIntercept " +
+      s"coefficients but got $coefficientSize")
+    require(numClasses <= 2, s"Binary logistic aggregator requires numClasses in {1, 2}" +
+      s" but found $numClasses.")
+  }
 
   private var weightSum = 0.0
   private var lossSum = 0.0
 
-  private val gradientSumArray =
-    Array.ofDim[Double](if (fitIntercept) numFeatures + 1 else numFeatures)
+  private val totalCoefficientLength = {
+    val cols = if (fitIntercept) numFeatures + 1 else numFeatures
+    val rows = if (multinomial) numClasses else 1
+    rows * cols
+  }
+
+  private val gradientSumArray = Array.ofDim[Double](totalCoefficientLength)
+
+  if (multinomial && numClasses < 2) {
```

Comment:
Is it `numClasses <= 2`? Also, in the doc, maybe you can say `Multinomial logistic (softmax) regression` and then ` When regularization is applied, multinomial (softmax) loss will produce a result different from binary loss since the positive and negative don't share the coefficients while the binary LOR shares the coefficients between positive and negative.`